dlt/extract/utils.py

```diff
@@ -119,6 +131,35 @@ def check_compat_transformer(name: str, f: AnyFun, sig: inspect.Signature) -> in
     return meta_arg
 
 
+def wrap_async_generator(
+    gen: AsyncGenerator[TDataItems, None]
+) -> Generator[Awaitable[TDataItems], None, None]:
+    """Wraps an async generator into a list of awaitables"""
+    exhausted = False
+    lock = asyncio.Lock()
+
+    # creates an awaitable that will return the next item from the async generator
+    async def run() -> TDataItems:
+        async with lock:
+            try:
+                return await gen.__anext__()
+            # on stop iteration mark as exhausted
+            except StopAsyncIteration:
+                nonlocal exhausted
+                exhausted = True
+                raise
+
+    # this generator yields None while the async generator is not exhausted
+    try:
+        while not exhausted:
+            while lock.locked():
+                yield None
+            yield run()
+    except GeneratorExit:
+        # clean up async generator
```

Comment:
i don't think I need to clean up here, this can happen at the end of the loop in the pipe

Reply:
the problem is, that the "parent" generator gets closed before the last item is evaulated that we want to evaluate. so it has to stay this way. do you know what I mean?

Reply:
But i just had a nice idea of how to close the generator in the right moment this way, which is a good improvement :)